src/routes/conversation/[id]/+server.ts

```diff
@@ -195,6 +195,14 @@ export async function POST({ request, locals, params, getClientAddress }) {
 					updates.push(newUpdate);
 				}
 				controller.enqueue(JSON.stringify(newUpdate) + "\n");
+
+				if (newUpdate.type === "finalAnswer") {
+					// 4096 of spaces to make sure the browser doesn't blocking buffer that holding the response
+					controller.enqueue(" ".repeat(4096));
+				}
+				if (newUpdate.type === "stream" && newUpdate.token === "") {
+					return;
```

Comment:
Maybe i'm missing something but I think the function will return anyway after this condition so I still don't know what this is supposed to do. Currently the function does the same thing whether or not `newUpdate.token === ""`

Should the check be above the `controller.enqueue(JSON.stringify(newUpdate) + "\n");`line so it returns early?

Reply:
As you mentioned, `newUpdate.token === "" `seems not work if it placed after `controller.enqueue(JSON.stringify(newUpdate) + "\n");`

So, I changed to above the 
```
if (newUpdate.type !== "stream") {
	updates.push(newUpdate);
}
```